tfjs-core/src/io/io_utils.ts

```diff
@@ -177,6 +178,8 @@ export function decodeWeights(
           const v = quantizedArray[i];
           values[i] = Math.round(v * quantization.scale + quantization.min);
         }
+      } else if (env().getBool('FLOAT16') && dtype === 'float16') {
```

Comment:
If https://github.com/tensorflow/tfjs/pull/7535/files#diff-7a8f5470570deed75c0bf9fbfeb2b942675c59fc60986222eab4ecdf39073241R491 is correctly set, does it mean we already checked `env().getBool('FLOAT16')`? If yes, `else if (dtype === 'float16') is enough?